app/controllers/application_controller.rb

```diff
@@ -73,6 +75,14 @@ def validate_user_profile!
     end
   end
 
+  def validate_archived_organization!
+    raise Mumuki::Domain::ArchivedOrganizationError if Organization.current.archived? && !current_user.teacher?
+  end
+
+  def validate_enabled_organization!
+    raise Mumuki::Domain::DisabledOrganizationError if Organization.current.disabled? && !current_user.teacher?
```

Comment:
Se are checking domain conditions, using domain objects and raising domain exceptions. Shouldn't we move this logic to domain, too?

Something like organization.validate_accessible_for(user)

I Even think there is already a method like that somewhere

Reply:
Yep, I agree.
I think that moving this logic to `validate_accessible!(item)` makes a lot of sense.